compiler/src/Language/Bond/Codegen/TypeMapping.hs

```diff
@@ -513,10 +513,10 @@ javaType BT_Blob = pure "byte[]"
 javaType (BT_IntTypeArg x) = pureText x
 javaType (BT_Maybe type_) = javaType (BT_Nullable type_)
 javaType (BT_Nullable element) = javaBox element
-javaType (BT_List element) = "java.util.LinkedList<" <>> javaBox element <<> ">"
-javaType (BT_Vector element) = "java.util.ArrayList<" <>> javaBox element <<> ">"
-javaType (BT_Set element) = "java.util.HashSet<" <>> javaBox element <<> ">"
-javaType (BT_Map key value) = "java.util.HashMap<" <>> javaBox key <<>> ", " <>> elementTypeName value <<> ">"
+javaType (BT_List element) = "java.util.List<" <>> javaBox element <<> ">"
+javaType (BT_Vector element) = "java.util.List<" <>> javaBox element <<> ">"
+javaType (BT_Set element) = "java.util.Set<" <>> javaBox element <<> ">"
+javaType (BT_Map key value) = "java.util.Map<" <>> javaBox key <<>> ", " <>> elementTypeName value <<> ">"
```

Comment:
There's no need to impose a particular `List` implementation on generated classes. If we initialize with `ArrayList`s and someone decides they'd rather work with a `LinkedList`, or some funky list of their own creation, serialization will work fine.

The original state was a mistake I made while trying to figure out how codegen works. I noticed it because I wanted to use a `TreeSet` and `TreeMap`, as you saw below. (Details on that are also below.)